helpers/questions_filter/questions_filter.py

```diff
@@ -0,0 +1,33 @@
+from dateutil import parser
+from dateutil.relativedelta import relativedelta
+from graphql import GraphQLError
+
+
+def filter_questions_by_date_range(query, start_date, end_date):
+    """
+    Return questions that  fall in the date range
+    """
+    if not (start_date and end_date):
+        return query
+    start_date, end_date = format_range_dates(start_date, end_date)
+    questions = []
+    for question in query:
+        question_start_date = parser.parse(question.start_date)
+        question_end_date = parser.parse(question.end_date)
+        if question_start_date >= start_date and question_end_date <= end_date:
+            questions.append(question)
+    return questions
+
+
+def format_range_dates(start_date, end_date):
```

Comment:
hello @ThaDeveloper, there is a function in the codebase called `validate_date_time_range` in `utitlities\validations.py` that does the exact implementation. Can we use that rather than creating another one

Reply:
@RachaelNantale The reason `validate_date_time_range` won't be suited in this case is that it only does validation which isn't much applicable when filtering questions, for instance, the condition below only applies when creating questions but we can get questions whose start_date is lower than today.  Also, `format_range_dates` convert the date format based on how we are storing data in the backend.

Reply:
@ThaDeveloper , so these functions are basically doing the same thing.  The one already in the codebase can changed to cater for  this as seen below 
```def validate_date_time_range(**kwargs):
    """
    Function to validate the dates entered
    for questions for feedback
    :params kwargs
    """
    if 'start_date' in kwargs and isinstance(kwargs['start_date'], str):
        kwargs['start_date'] = datetime.datetime.strptime(kwargs['start_date'], '%Y %b %d')
        kwargs['end_date'] = datetime.datetime.strptime(kwargs['end_date'], '%Y %b %d')
    if ('start_date' and 'end_date' in kwargs) and\
            kwargs['start_date'] < datetime.datetime.now():
        raise ValueError('startDate should be today or after')
    elif ('start_date' and 'end_date' in kwargs) and\
            kwargs['end_date'] - kwargs['start_date'] < datetime.timedelta(
                days=1
            ):
        raise ValueError(
            'endDate should be at least a day after startDate'
        )
```

Reply:
Hello @ThaDeveloper , If you need more light on the issue raised by @RachaelNantale , Kindly ping me for assistance. Otherwise, that issue raised is valid.

Reply:
@nakatuddesuzan I just had sync with her, there's something she's confirming. Thanks